samcli/lib/list/stack_outputs/stack_outputs_producer.py

```diff
@@ -0,0 +1,61 @@
+"""
+The producer for the 'sam list stack-outputs' command
+"""
+import dataclasses
+import logging
+
+from botocore.exceptions import ClientError, BotoCoreError
+from samcli.commands.list.exceptions import SamListError, NoOutputsForStackError, StackDoesNotExistInRegionError
+
+from samcli.lib.list.list_interfaces import Producer
+from samcli.lib.list.stack_outputs.stack_outputs import StackOutputs
+
+LOG = logging.getLogger(__name__)
+
+
+class StackOutputsProducer(Producer):
+    def __init__(self, stack_name, output, region, cloudformation_client, mapper, consumer):
+        self.stack_name = stack_name
+        self.output = output
+        self.region = region
+        self.cloudformation_client = cloudformation_client
+        self.mapper = mapper
+        self.consumer = consumer
+
+    def get_stack_info(self):
+        """
+        Returns the stack output information for the stack and raises exceptions accordingly
+
+        Returns
+        -------
+            A dictionary containing the stack's information
+        """
+
+        try:
+            response = self.cloudformation_client.describe_stacks(StackName=self.stack_name)
+            if not response.get("Stacks", []):
+                raise StackDoesNotExistInRegionError(stack_name=self.stack_name, region=self.region)
+            if len(response.get("Stacks", [])) > 0 and "Outputs" not in response.get("Stacks", [])[0]:
+                raise NoOutputsForStackError(stack_name=self.stack_name, region=self.region)
+            return response["Stacks"][0]["Outputs"]
+
+        except ClientError as e:
+            if "Stack with id {0} does not exist".format(self.stack_name) in str(e):
```

Comment:
You can use this method to get the error code and compare against it: https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/utils/boto_utils.py#L140

It would be better since exception message might change but it is highly unlikely that the error code will change.